aws_topology/stackstate_checks/aws_topology/resources/route53_hostedzone.py

```diff
@@ -1,40 +1,103 @@
+
+    def collect_hosted_zone(self, hosted_zone_data):
+        hosted_zone_id = hosted_zone_data.get("Id", "")
+        data = self.collect_hosted_zone_description(hosted_zone_id) or self.construct_hosted_zone_description(
+            hosted_zone_data
+        )
+        tags = self.collect_tags(hosted_zone_id) or []
+        resource_record_sets = self.collect_resource_record_sets(hosted_zone_id) or []
+        return HostedZoneData(hosted_zone=data, tags=tags, resource_record_sets=resource_record_sets)
+
+    def collect_hosted_zones(self):
+        for hosted_zone in [
+            self.collect_hosted_zone(hosted_zone_data)
+            for hosted_zone_data in client_array_operation(self.client, "list_hosted_zones", "HostedZones")
+        ]:
+            yield hosted_zone
+
+    @set_required_access_v2("route53:ListHostedZones")
+    def process_hosted_zones(self):
+        for hosted_zone_data in self.collect_hosted_zones():
+            self.process_hosted_zone(hosted_zone_data)
+
     def process_all(self, filter=None):
         """
         Route 53 hosted zones contain DNS records. A AWS Domain can point to a hosted zone.
         """
-        for list_hosted_zones_page in self.client.get_paginator('list_hosted_zones').paginate():
-            for hosted_zone in list_hosted_zones_page.get('HostedZones') or []:
-                self.process_hosted_zone(hosted_zone)
-
-    def process_hosted_zone(self, hosted_zone):
-        hosted_zone_data = {}
-        hosted_zone_id = hosted_zone['Id']
-        resource_id = hosted_zone_id.rsplit('/', 1)[-1]
-        tags = self.client.list_tags_for_resource(
-            ResourceType='hostedzone',
-            ResourceId=resource_id
-        ).get('ResourceTagSet')
-        if tags:  # pragma: no cover
-            hosted_zone_data['Tags'] = tags.get('Tags') or []
-        hosted_zone_data['Id'] = hosted_zone_id
-        hosted_zone_detail_raw = self.client.get_hosted_zone(Id=hosted_zone_id)
-        hosted_zone_detail = make_valid_data(hosted_zone_detail_raw)
-        hosted_zone_data['HostedZone'] = hosted_zone_detail['HostedZone']
-        if 'DelegationSet' in hosted_zone_detail:  # pragma: no cover
-            hosted_zone_data['DelegationSet'] = hosted_zone_detail['DelegationSet']
-
-        resource_record_sets = self.client.list_resource_record_sets(HostedZoneId=hosted_zone_id)
-        hosted_zone_data['ResourceRecordSets'] = resource_record_sets.get('ResourceRecordSets') or []
-        hosted_zone_data['URN'] = [
-            "arn:aws:route53:::{}".format(hosted_zone_id.lstrip('/'))
+        if not filter or "hostedzones" in filter:
+            self.process_hosted_zones()
+
+    @transformation()
+    def process_hosted_zone(self, data):
+        hosted_zone = HostedZone(data.hosted_zone, strict=False)
+        hosted_zone.validate()
+        output = make_valid_data(data.hosted_zone)
+        hosted_zone_id = hosted_zone.HostedZone.Id.rsplit("/", 1)[-1]
```

Comment:
Can this potentially break?

Reply:
Unlikely - if a value is not found, Schematics has a default value of `UNKNOWN`. If no forward slash is found, then it will return the whole string as usual

Reply:
The reason for this is the format is `/hostedzone/<id of resource>` - this just removes the first section